cas-server-support-saml/src/main/java/org/jasig/cas/support/saml/web/flow/mdui/StaticMetadataResolverAdapter.java

```diff
@@ -0,0 +1,119 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.support.saml.web.flow.mdui;
+
+import org.opensaml.saml.metadata.resolver.filter.impl.MetadataFilterChain;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SchedulerFactory;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.springframework.core.io.Resource;
+
+import javax.annotation.PostConstruct;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * A {@link StaticMetadataResolverAdapter} that loads metadata from static xml files
+ * served by urls or locally.
+ *
+ * @author Misagh Moayyed
+ * @since 4.1.0
+ */
+public final class StaticMetadataResolverAdapter extends AbstractMetadataResolverAdapter implements Job {
+    private static final int DEFAULT_METADATA_REFRESH_INTERNAL_MINS = 300;
+
+    /**
+     * Refresh metadata every {@link #DEFAULT_METADATA_REFRESH_INTERNAL_MINS}
+     * minutes by default.
+     **/
+    private int refreshIntervalInMinutes = DEFAULT_METADATA_REFRESH_INTERNAL_MINS;
+
+    /**
+     * New ctor - required for serialization and job scheduling.
+     */
+    public StaticMetadataResolverAdapter() {
+        super();
+    }
+
+    /**
+     * Instantiates a new static metadata resolver adapter.
+     *
+     * @param metadataResources the metadata resources
+     */
+    public StaticMetadataResolverAdapter(final Map<Resource, MetadataFilterChain> metadataResources) {
+        super(metadataResources);
+    }
+
+    public void setRefreshIntervalInMinutes(final int refreshIntervalInMinutes) {
+        this.refreshIntervalInMinutes = refreshIntervalInMinutes;
+    }
+
+    @Override
+    protected InputStream getResourceInputStream(final Resource resource, final String entityId) throws IOException {
+        if (!resource.exists() || !resource.isReadable()) {
+            throw new FileNotFoundException("Resource does not exist or is unreadable");
+        }
+        return resource.getInputStream();
+    }
+
+    /**
+     * Refresh metadata. Schedules the job to retrieve metadata.
+     * @throws SchedulerException the scheduler exception
+     */
+    @PostConstruct
+    public void refreshMetadata() throws SchedulerException {
+        final Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                buildMetadataResolverAggregate(null);
+            }
+        });
+        thread.start();
+
+        final JobDetail job = JobBuilder.newJob(this.getClass())
```

Comment:
I reviewed this once more; I don't think we'd be able to move this to the XML config because it's a feature of the component that should automatically start as soon as CAS is started. It's not an add-on that can be optional; it must exist and I wouldnt want the adopter to create a whole bunch of complicated XML to manage this reloading aspect.